### PR TITLE
Compiling fixes

### PR DESCRIPTION
**Pull request by otaviobp**
- Remove malloc/free calls from cborparser
 - Remove warnings from systems with glibc >= 2.2
 - Updates in .gitignore

### CODE REVIEW

File: src/cborparser_dup_string.c [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+** LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+** OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+** THE SOFTWARE.
+**
+****************************************************************************/
+
+#define _BSD_SOURCE 1
+#include "cbor.h"
+#include <stdlib.h>
+//#include "cborconstants_p.h"
```
Comment (thiagomacieira) [inline src/cborparser_dup_string.c]: Please remove these unnecessary commented out lines.

**Comment by thiagomacieira** [comment]
Looks good, aside from the extra commented out lines that don't serve a purpose. Can you remove them?

**Comment by otaviobp** [comment]
Sorry. I have fixed that,.

**Comment by thiagomacieira** [comment]
Merged as e2d5dd5

**Pull request closed without merging**